src/Tokenizer/Tokens.php

```diff
@@ -1348,7 +1436,7 @@ private function registerFoundToken($token): void
     /**
      * Register token as found.
      *
-     * @param array{int}|string|Token $token token prototype
+     * @param array{0: int, 1?: string}|string|Token $token token prototype
```

Comment:
those changes could go potentially as separated PR, without having all topics concluded, thus those changes could effectively be merged sooner

Reply:
@julienfalque , I would love to break this PR into smaller parts. Too big change at once.

Let's pick a next/prev method you want to change, and let's see if we can drop usage of old-styled method in our own repo, and if we can, let's merge that one first.
Then we will have the knowledge how it should look, like single/multi-token kind needle thingy.

then we look to further methods